src/components/signup/SignUp.js

```diff
@@ -0,0 +1,181 @@
+import React, { Component } from 'react';
+import logo from '../../images/logo.png';
+import { Link } from 'react-router-dom';
+import './SignUp.css';
+
+const emailRegex = RegExp(/^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/);
+//Add the telephone regex
+const formValid = ({ formErrors, ...rest }) => {
+  let valid = true;
+
+  // validate form errors being empty
+  Object.values(formErrors).forEach((val) => {
+    val.length > 0 && (valid = false);
+  });
+
+  // validate the form was filled out
+  Object.values(rest).forEach((val) => {
+    val === null && (valid = false);
+  });
+
+  return valid;
+};
+
+class Login extends Component {
```

Comment:
@morelmiles the class name doesn't match the implemented work, if possible let us match our class names to the file name

Reply:
this has been handled